core/src/impl/Kokkos_MemorySpace.hpp

```diff
@@ -55,12 +55,15 @@
 #include <impl/Kokkos_Error.hpp>
 
 #include <string>
-#include <iostream>
-#include <sstream>
 
 namespace Kokkos {
 namespace Impl {
 
+// Defined in implementation file to avoid having to
+void safe_throw_allocation_with_header_failure(
```

Comment:
Why "safe"?

Reply:
because it allocates space for the error message and then checks to make sure that allocation didn't itself throw, and so on.